File: jdroid-android-sample/src/debug/java/com/jdroid/android/sample/debug/AndroidDebugContext.java
package com.jdroid.android.sample.debug;

import androidx.core.util.Pair;

import com.jdroid.android.debug.DebugContext;
import com.jdroid.android.sample.api.ApiServer;
import com.jdroid.android.sample.firebase.remoteconfig.AndroidRemoteConfigParameter;
import com.jdroid.java.collections.Lists;
import com.jdroid.java.collections.Maps;
import com.jdroid.java.http.Server;
import com.jdroid.java.remoteconfig.RemoteConfigParameter;

import java.util.Arrays;
import java.util.List;
import java.util.Map;

public class AndroidDebugContext extends DebugContext {

	public AndroidDebugContext() {
		addCustomDebugInfoProperty(new Pair<String, Object>("Sample Key", "Sample Value"));
	}

	@Override
	public Map<Class<? extends Server>, List<? extends Server>> getServersMap() {
		Map<Class<? extends Server>, List<? extends Server>> serversMap = Maps.newHashMap();
		serversMap.put(ApiServer.class, Lists.newArrayList(ApiServer.values()));
		return serversMap;
	}

	@Override
	public List<String> getUrlsToTest() {
		List<String> urls = Lists.newArrayList();
		urls.add("http://jdroidtools.com");
		urls.add("http://jdroidtools.com/");
		urls.add("http://jdroidtools.com/uri");
		urls.add("http://jdroidtools.com/uri/singletop?a=1");
		urls.add("http://jdroidtools.com/uri/singletop?a=2");
		urls.add("http://jdroidtools.com/uri/singletop?a=3");
		urls.add("http://jdroidtools.com/uri/noflags?a=1");
		urls.add("http://jdroidtools.com/uri/noflags?a=2");
		urls.add("http://jdroidtools.com/uri/noflags?a=3");
		urls.add("http://jdroidtools.com/uri/noflags");
		urls.add("http://jdroidtools.com/uri/invalid");
		return urls;
	}

	@Override
	public List<RemoteConfigParameter> getRemoteConfigParameters() {
		return Arrays.asList(AndroidRemoteConfigParameter.values());
	}
}

File: jdroid-android-sample/src/debug/java/com/jdroid/android/sample/debug/AndroidDebugContext.kt
package com.jdroid.android.sample.debug

import androidx.core.util.Pair

import com.jdroid.android.debug.DebugContext
import com.jdroid.android.sample.api.ApiServer
import com.jdroid.android.sample.firebase.remoteconfig.AndroidRemoteConfigParameter
import com.jdroid.java.collections.Lists
import com.jdroid.java.collections.Maps
import com.jdroid.java.http.Server
import com.jdroid.java.remoteconfig.RemoteConfigParameter

import java.util.Arrays

class AndroidDebugContext : DebugContext() {
    init {
        addCustomDebugInfoProperty(Pair("Sample Key", "Sample Value"))
    }

    override fun getServersMap(): Map<Class<out Server>, List<Server>> {
        val serversMap = Maps.newHashMap<Class<out Server>, List<Server>>()
        serversMap[ApiServer::class.java] = Lists.newArrayList(*ApiServer.values())
        return serversMap
    }

    override fun getUrlsToTest(): List<String> {
        val urls = Lists.newArrayList<String>()
        urls.add("http://jdroidtools.com")
        urls.add("http://jdroidtools.com/")
        urls.add("http://jdroidtools.com/uri")
        urls.add("http://jdroidtools.com/uri/singletop?a=1")
        urls.add("http://jdroidtools.com/uri/singletop?a=2")
        urls.add("http://jdroidtools.com/uri/singletop?a=3")
        urls.add("http://jdroidtools.com/uri/noflags?a=1")
        urls.add("http://jdroidtools.com/uri/noflags?a=2")
        urls.add("http://jdroidtools.com/uri/noflags?a=3")
        urls.add("http://jdroidtools.com/uri/noflags")
        urls.add("http://jdroidtools.com/uri/invalid")
        return urls
    }

    override fun getRemoteConfigParameters(): List<RemoteConfigParameter> {
        return Arrays.asList<RemoteConfigParameter>(*AndroidRemoteConfigParameter.values())
    }
}
